tools/zstd/Makefile

```diff
@@ -11,25 +11,33 @@ PKG_LICENSE:=BSD-3-Clause
 PKG_LICENSE_FILES:=LICENSE
 PKG_CPE_ID:=cpe:/a:facebook:zstandard
 
-include $(INCLUDE_DIR)/host-build.mk
-include $(INCLUDE_DIR)/meson.mk
+HOST_BUILD_PARALLEL:=1
 
-MESON_HOST_BUILD_DIR:=$(HOST_BUILD_DIR)/build/meson/openwrt-build
+include $(INCLUDE_DIR)/host-build.mk
 
 HOSTCC:= $(HOSTCC_NOCACHE)
 
-MESON_HOST_ARGS += \
-	-Ddefault_library=static \
-	-Dlegacy_level=7 \
-	-Ddebug_level=0 \
-	-Dbacktrace=disabled \
-	-Dstatic_runtime=false \
-	-Dbin_programs=true \
-	-Dbin_tests=false \
-	-Dbin_contrib=false \
-	-Dmulti_thread=enabled \
-	-Dzlib=disabled \
-	-Dlzma=disabled \
-	-Dlz4=disabled
+HOST_MAKE_FLAGS += \
+	ZSTD_LEGACY_SUPPORT=7 \
```

Comment:
BTW, why are we doing this?
From https://github.com/facebook/zstd/blob/dev/lib/README.md#modular-build
> By default, this build macro is set as ZSTD_LEGACY_SUPPORT=5

Reply:
Current meson build does the same, I guess its just ported straight from meson options

Reply:
so that means it can probably be dropped then, like the non-parallel build :)
I think we really should avoid diverging from upstream defaults unless we have a very good technical reason for doing so, imho.

Reply:
I agree, the ZSTD_LEGACY_SUPPORT=7 seems to appeared during 1.5.2 update and meson conversion.

Reply:
I did this as legacy support is not needed. I don’t remember if it shaved a little binary space.

Reply:
I don't think "shaving a little binary space" is very relevant here. Perfect compatibility with upstream defaults seems far more important imho.

Reply:
Honestly that option just disables support for old incompatible formats... Something that should never happen since nobody should generate stuff from zstd 5.0.

For upstream they use 5 to support ancient archive but this is not the case on OpenWRT... Honestly I would even set it to 0 considering how new the zstd adoption is...

We would use it only for our mirrors generated tar so the incompatibility problem doesn't apply. But I'm open on this... It's really just taste.

Reply:
it's disabling the ability to *read* these older formats as well, meaning the ability to read potentially older archives.

But again the more important point is that we're diverging from vanilla upstream and I think this is something we should really avoid for core tools. It means we'd be using a potentially less tested (covered) variant that's special to *us* only, and I don't think it's a very good idea. My 2c.

Reply:
I kind of have to agree with @f00b4r0 for host tools it doesn't really make sense to save on couple of kB in this case.